recipes/pipewire/all/conanfile.py

```diff
@@ -0,0 +1,431 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import (
+    apply_conandata_patches,
+    copy,
+    export_conandata_patches,
+    get,
+    load,
+    rm,
+    rmdir,
+    save,
+)
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.64.0 <2 || >=2.2.0"
+
+
+class PipeWireConan(ConanFile):
+    name = "pipewire"
+    description = (
+        "PipeWire is a server and user space API to deal with multimedia pipelines."
+    )
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://pipewire.org/"
+    topics = ("audio", "graph", "pipeline", "stream", "video")
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "flatpak": [True, False],
+        "gsettings": [True, False],
+        "raop": [True, False],
+        "with_avahi": [True, False],
+        "with_dbus": [True, False],
+        "with_ffmpeg": [True, False],
+        "with_libalsa": [True, False],
+        "with_libsndfile": [True, False],
+        "with_libudev": [True, False],
+        "with_ncurses": [True, False],
+        "with_opus": [True, False],
+        "with_pulseaudio": [True, False],
+        "with_readline": [True, False],
+        "with_selinux": [True, False],
+        "with_vulkan": [True, False],
+        "with_x11": [True, False],
+        "with_xfixes": [True, False],
+    }
+    default_options = {
+        "flatpak": True,
+        "gsettings": True,
+        "raop": True,
+        "with_avahi": True,
+        "with_dbus": True,
+        "with_ffmpeg": False,
+        "with_libalsa": True,
+        "with_libsndfile": True,
+        "with_libudev": True,
+        "with_ncurses": True,
+        "with_opus": True,
+        "with_pulseaudio": True,
+        "with_readline": True,
+        "with_selinux": True,
+        "with_vulkan": True,
+        "with_x11": True,
+        "with_xfixes": True,
+    }
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if os.getenv("CONAN_CENTER_BUILD_SERVICE") is not None:
+            self.options.with_xfixes = False
```

Comment:
A source-code comment explaining the purpose of this would be nice.

Reply:
Done.

Reply:
Thanks!